_docs/developer/testing/cypress.md

```diff
@@ -0,0 +1,187 @@
+
+## Installation 
+
+Make sure you have `node.js` version 10 or higher installed locally. You can check using the command: `node -v` 
+
+```bash
+# from the site directory
+npm install --include=dev
+```
+
+If you are on Linux you may need the following dependencies if they are not already installed:
+```bash
+apt-get install libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+```
+
+## Cypress Test Runner
+
+You can open the test runner using the command:
+
+```bash
+# from the site directory
+npx cypress open
+```
+
+*Note:* On startup Cypress will try and check if the site is running on the `http://localhost:1501` address, make sure you have a local instance of Submitty that is running while testing with Cypress. 
+
+### Running Tests
+
+From the test runner you should see a list of spec files. You can either run everything by clicking "Run integration spec(s)" on the top right corner or click on an individual file to only run the tests under that file. You can also choose which browser installed on your machine to run the tests against. 
+
+![Example interface of running spec file](/images/test-running-interface.png) 
+
+Once the tests have been run you can interact with the site on the right hand side of the runner. This is a good way of debugging what state the site is in if a test fails. You can also right click and see the console for any errors or information. During this time you're terminal should have a list of HTTP responses the site has been sending to your browser. For example:
+
+*Example Requests*
+```
+HEAD / - - ms - -
+HEAD / - - ms - -
+HEAD / - - ms - -
+POST /authentication/check_login?old=http%253A%252F%252Flocalhost%253A1501%252Fhome 302 287.491 ms - 96
+GET /home 200 287.217 ms - -
+GET /authentication/login 200 2.297 ms - -
+POST /authentication/check_login?old=http%253A%252F%252Flocalhost%253A1501%252Fhome 302 2772.070 ms - 96
+GET /home 302 210.507 ms - 210
+GET /authentication/login?old=http%3A%2F%2Flocalhost%3A1501%2Fhome 200 264.544 ms - -
+GET /authentication/login 200 1.629 ms - -
+POST /authentication/check_login?old=http%253A%252F%252Flocalhost%253A1501%252Fhome 302 205.710 ms - 96
+```
+
+A complete overview can be found [here](https://docs.cypress.io/guides/core-concepts/test-runner.html#Overview)
+
+### Running Headless
+
+You can run the entire test suite in headless mode. This will pipe the browser display directly to Cypress without getting displayed and is faster than the test-runner. This can be done with the command:
+
+```bash
+# from the site directory
+npx cypress run 
+```
+
+If any test fails a screen shot will be taken of the step that failed. Additionally, a video will be recorded each time of the run. They can be found under the `site/cypress/screenshots` and `site/cypress/videos` directories.
+
+See the full usage [here](https://docs.cypress.io/guides/tooling/plugins-guide.html#Use-Cases)
+
+## Writing Tests
```

Comment:
Worth linking to https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests.html#Folder-Structure?